climada/engine/unsequa/calc_cost_benefit.py

```diff
@@ -168,52 +179,81 @@ def uncertainty(self, unc_data, pool=None, **cost_benefit_kwargs):
         See Also
         --------
         climada.engine.cost_benefit:
-            Compute risk and adptation option cost benefits.
+            compute risk and adptation option cost benefits.
 
         """
 
-        if unc_data.samples_df.empty:
+        if unc_sample.samples_df.empty:
             raise ValueError("No sample was found. Please create one first" +
                         "using UncImpact.make_sample(N)")
 
-        samples_df = unc_data.samples_df.copy(deep=True)
+        chunksize = np.ceil(
+            unc_sample.samples_df.shape[0] / processes
+            ).astype(int) if chunksize is None else chunksize
+
+        samples_df = unc_sample.samples_df.copy(deep=True)
         unit = self.value_unit
 
         LOGGER.info("The freq_curve is not saved. Please "
                     "change the risk_func (see climada.engine.cost_benefit) "
                     "if return period information is needed")
 
         start = time.time()
-        one_sample = samples_df.iloc[0:1].iterrows()
-        cb_metrics = map(self._map_costben_calc, one_sample)
+        one_sample = samples_df.iloc[0:1]
+        p_iterator = self._sample_parallel_iterator(
+                one_sample,
+                chunksize=chunksize,
+                ent_input_var=self.ent_input_var,
+                haz_input_var=self.haz_input_var,
+                ent_fut_input_var=self.ent_fut_input_var,
+                haz_fut_input_var=self.haz_fut_input_var,
+                cost_benefit_kwargs=cost_benefit_kwargs
+            )
+        cb_metrics = itertools.starmap(_map_costben_calc, p_iterator)
         [imp_meas_present,
          imp_meas_future,
          tot_climate_risk,
          benefit,
-         cost_ben_ratio] = list(zip(*cb_metrics))
+         cost_ben_ratio] = [
+                 list(itertools.chain.from_iterable(x))
+                 for x in list(zip(*cb_metrics))
+                 ]
         elapsed_time = (time.time() - start)
-        self.est_comp_time(unc_data.n_samples, elapsed_time, pool)
+        self.est_comp_time(unc_sample.n_samples, elapsed_time, processes)
 
         #Compute impact distributions
         with log_level(level='ERROR', name_prefix='climada'):
-            if pool:
-                LOGGER.info('Using %s CPUs.', pool.ncpus)
-                chunksize = max(min(unc_data.n_samples // pool.ncpus, 100), 1)
-                cb_metrics = pool.map(partial(self._map_costben_calc, **cost_benefit_kwargs),
-                                      samples_df.iterrows(),
-                                      chunksize=chunksize)
-
+            p_iterator = self._sample_parallel_iterator(
```

Comment:
Done in 261415ceaacd2ff8d7146e4a1d372509c1f925c6

Reply:
With "local" I meant a function that is defined (only) in the scope of `uncertainty`. This way, it would need fewer parameters. But it works this way nonetheless